finality-aleph/src/aggregator/multicast.rs

```diff
@@ -0,0 +1,93 @@
+//! A set of abstractions for dealing with `ReliableMulticast` in a more testable
+//! and modular way.
+//!
+//! We expose the following traits:
+//! - `Multicast`: mimicking the interface of `aleph_bft::ReliableMulticast`
+//! - `Multisigned`: abstracting away the `aleph_bft::Multisigned` capabilities as a trait.
+
+use crate::crypto::Signature;
+use aleph_bft::rmc::ReliableMulticast;
+use aleph_bft::{
+    rmc::Message, MultiKeychain, Multisigned as MultisignedStruct, Signable, SignatureSet,
+    UncheckedSigned,
+};
+use codec::{Codec, Decode, Encode};
+use sp_runtime::traits::Block;
+use std::{fmt::Debug, hash::Hash as StdHash};
+
+/// A wrapper allowing block hashes to be signed.
+#[derive(PartialEq, Eq, StdHash, Clone, Debug, Default, Encode, Decode)]
+pub struct SignableHash<H: Codec + Send + Sync> {
```

Comment:
I do not feel so comfortable with this design, that here we require from `H` some properties that seems to be unnecessary (`Signable` expects `H` to be only `AsRef<[u8]>`) and nearby we define a trait `Hash` with a superset of these properties; I would either require minimal set here, or just use this `Hash` everywhere

Also, the trait `Hash` seems it is needed only here and in `BlockSignatureAggregator`, so probably it can be moved to `mod.rs` and stay private.

Reply:
I went for using the `Hash` everywhere here, so for now I can't make it private as it would leak as part of the public `SignableHash` definition. In a subsequent refactor I might strip down the type bounds to the bare minimum and then this refactor would be possible.

What do you think?

Reply:
I went with the following structure:
```
+ aggregation/
|- aggregator.rs
|- mod.rs
|- multicast.rs
```

Reply:
> I went for using the Hash everywhere here...
for me it is completely fine; further refactor with limiting bounds is not so urgent as long as we do not intend to mock it or implement it somewhere outside
> I went with the following structure...
great, beautiful! :smiley: